src/ophys_etl/transforms/suite2p_wrapper.py

```diff
@@ -0,0 +1,249 @@
+                         "subtraction routine."))
+    min_neuropil_pixels = argschema.fields.Int(
+            default=350,
+            description=("Minimum number of pixels in the neuropil."
+                         "Allen production will run it's own neuropil "
+                         "subtraction routine."))
+    allow_overlap = argschema.fields.Bool(
+            # NOTE: changing this from extraction validation and labeling
+            # will be untangled downstream in demixing.
+            default=True,
+            description=("Pixels that are overlapping are thrown out (False) "
+                         "or added to both ROIs (True). Allen production will "
+                         "run a demixing step that should allow both ROIs "
+                         "to share pixels."))
+    # Allen-specific options
+    bin_size = argschema.fields.Int(
+            required=False,
+            description=("If nbinned not provided, will calculate nbinned as "
+                         "nframes / bin_size. This allows consistent temporal "
+                         "downsampling across movies of different lengths."))
+    output_dir = argschema.fields.OutputDir(
+            required=True,
+            description="for minimal and cleaner output, specifies output dir")
+    retain_files = argschema.fields.List(
+            argschema.fields.Str,
+            cli_as_single_argument=True,
+            required=True,
+            default=['stat.npy'],
+            description=("only Suite2P output files with basenames in this "
+                         "list will be retained. If 'all', a complete list "
+                         "will be retained."))
+    timestamp = argschema.fields.Bool(
+            default=True,
+            description=("if true, will add a timestamp to the output file "
+                         "names. "
+                         "<basename>.<ext> -> <basename>_<timestamp>.<ext>"))
+
+    @mm.post_load
+    def check_args(self, data, **kwargs):
+        if ('nbinned' not in data) & ('bin_size' not in data):
+            raise Suite2PWrapperException(
+                    "must provide either `nbinned` or `bin_size`")
+        return data
+
+    @mm.post_load
+    def check_retain_files(self, data, **kwargs):
+        if 'all' in data['retain_files']:
+            data['retain_files'] = [
+                    'ops1.npy', 'data.bin', 'Fneu.npy', 'F.npy', 'iscell.npy',
+                    'ops.npy', 'spks.npy', 'stat.npy']
+        return data
+
+
+class Suite2PWrapperOutputSchema(argschema.schemas.DefaultSchema):
+    output_files = argschema.fields.Dict(
+        keys=argschema.fields.Str,
+        values=argschema.fields.InputFile,
+        required=True,
+        description="retained output files from Suite2P")
+
+
+def copy_and_add_uid(srcdir: pathlib.Path, dstdir: pathlib.Path,
+                     basenames: List[str], uid: str = None) -> List[str]:
```

Comment:
Should be: `uid: Optional[str] = None` (include `from typing import Optional`)

Reply:
done